Add connection manager

matthiasbeyer opened the pull request:
This is my [WIP] proposal for the connection manager. I'd like to have a first review on this.

Targets #542.

matthiasbeyer committed:
Add connection manager interface+stub

matthiasbeyer committed:
Rename "connection_manager" type to "connection_processor" type

This was really confusing...

matthiasbeyer committed:
Export the implementation of the ws_connection_processor

matthiasbeyer committed:
Add implementation of create_connection_cb() callback

matthiasbeyer committed:
Change visibility of members of ws_connection_processor

neithernut commented on src/connection/config_file.c:
```diff
@@ -88,8 +88,8 @@ ws_connection_loadconf(void) {
-    struct ws_connection_manager* proc;
-    proc = ws_connection_manager_new(fd, deserializer, NULL);
+    struct ws_connection_processor* proc;
+    proc = ws_connection_processor_new(fd, deserializer, NULL);
```
This should be done via the _real_ connection _manager_.
You'll obviously need a method to do this.

matthiasbeyer replied:
What exactly do you mean by "this"? The whole configuration loading?

neithernut replied:
The contruction of a connection for the config.

matthiasbeyer replied:
You would add this to the scope of this PR? I would consider this as stuff for a second PR, as it is mostly refactoring work...

neithernut replied:
Nope, you just have to exrchange a function call or two.

neithernut commented on src/connection/manager.c:
```diff
@@ -25,5 +25,113 @@
+#include "util/socket.h"
+
+//!< @todo no hardcoded socket?
+#define SOCK_NAME "waysome.sock"
```
The socket name will _always_ be hardcoded (or, at least a default value will be).

matthiasbeyer replied:
I will remove the `@todo`.

neithernut commented:
You are missing deinitialization of the manager.

matthiasbeyer committed:
Add deinit-functionality to connection manager

matthiasbeyer committed:
Add function for manual creation of connection to fd

This patch creates a function in the connection manager to create a
connection to a fd manually.

It rewrites the creation-callback to use this function, to prevent
code-duplication.

matthiasbeyer committed:
Rewrite ws_connection_loadconf() to use ws_connection_manager_open_connection()

matthiasbeyer commented:
@neithernut Please review if this is what you wanted me to do (the latest 2 commits).

neithernut commented on src/connection/config_file.c:
```diff
@@ -76,25 +76,17 @@ open_config(void);
 int
 ws_connection_loadconf(void) {
-    int fd = open_config();
+    int fd = ws_connection_manager_init();
```
Please, at least rename `fd` to `res`.

neithernut commented:
Although the config manager is initialized on config load, you should probably initialize it explicitely in `main()` for good meassure.

matthiasbeyer committed:
Initialize the connection manager in main()

matthiasbeyer commented:
You mean the connection manager, but yes.

neithernut commented:
ACK.
